### PR TITLE
Ensure we have a post object in vip_legacy_redirect_parent_id

### PR DESCRIPTION
**Pull request by kjbenk**
Fixes issue where a post ID is passed to the `vip_legacy_redirect_parent_id` function when it is expecting an object.

### CODE REVIEW

File: wpcom-legacy-redirector.php [hunk 1]
```diff
@@ -455,6 +455,9 @@ public static function vip_legacy_redirect_parent_id( $post ) {
 				return true;
 			}
 		} else {
+			// Ensure we have a post object.
+			$post = get_post( $post );
```
Comment (rebeccahum) [inline wpcom-legacy-redirector.php]: Before calling `get_post()` unnecessarily, we should check that it's not an object.

**Review by rebeccahum — CHANGES_REQUESTED** [review]

**Comment by kjbenk** [comment]
@rebeccahum I looked into this issue some more and noticed that the real cause was from a `$_POST` var not being set during the CLI command.

https://github.com/Automattic/WPCOM-Legacy-Redirector/blob/master/wpcom-legacy-redirector.php#L453

I update the PR to ensure that during CLI command runs for the `import-from-meta` the proper vars are set.

**Comment by GaryJones** [comment]
> $_POST var not being set during the CLI command.

Is that even a thing? I can't see any other references to `$_POST` in that file, so I'm not sure why that's there.

The method got [moved](https://github.com/Automattic/WPCOM-Legacy-Redirector/commit/704b52da223554f1ff29b09f1eeda63c415f8145) from the new UI class, but in doing so, this `$_POST` suddenly appeared with no explanation!

**Comment by emrikol** [comment]
I think this will work for now.  It would be better if we didn't have to fiddle with the global vars, but that sounds like a bigger refactor.

**Pull request merged**